### ISSUE TITLE
Corrupt metadata error

### ISSUE THREAD

**Issue by unixfy** [andreimarcu/linx-server#221]
I migrated all of my data from Amazon S3 to a self-hosted Minio server, and I migrated the linx-server machine to a new provider as well. I backed up the files/, meta/, and config.ini files along with the contents of the S3 bucket and transferred everything to the new destinations with no problem.

However, when I try to access files that were uploaded before the migration occurred, I get the error "Corrupt Metadata". Is there any way I can "recover"/"fix" the metadata and restore access to the files?

Thanks.

**Comment by andreimarcu** [comment]
Could this be a permission issue? Do the old files have different users/permissions?
Other than that, if you open any file in the meta folder with a text editor, are you able to see the json object? Does anything look off there?

**Comment by mutantmonkey** [comment]
When using the S3 backend, the file metadata is stored as a tag alongside the object. If you still have access to the original bucket, there might be a tool that copy the tags over. If not, we'd need a tool that can regenerate the metadata similar to what is mentioned here: https://github.com/andreimarcu/linx-server/issues/203#issuecomment-605527879

**Comment by unixfy** [comment]
Permissions seem to be fine, and the meta folder is empty on both source and destination.

I think the tags is the issue here. When I was transferring the data I downloaded the files to disk and then reuploaded them to the new S3 bucket (which probably destroyed the metadata). Luckily I did not delete the old bucket so I will try to copy again while keeping tags intact.

Thanks mutantmonkey - this seems like something that should be documented ;)
